### PR TITLE
core syntax documentation copy edits and revisions

### PR DESCRIPTION
**Pull request by olets**
I looked at the v2 syntax documentation a few weeks ago, and found it hard to follow. Today I worked through it, changing things to help myself understand. This PR has all those changes.

I'm afraid it's all one big commit — I started at the top and read through. It's mostly:

- add a table of the transition properties
- add "transitions" to the heading names of things under Transitions, to help keep track of the section heirarchy
- move the `data` docs out from under Transitions
- some copy edits

There's a risk that this will feel inappropriately big! I know I'm new to this project, don't hesitate to say no to some or even all of it.

Thanks for a cool tool, I'm excited to try it out now that I get how it works 🎉

### CODE REVIEW

**Comment by thierrymichel** [comment]
Hi @olets,

Thanks for contributing!!!
We are aware that documentation can/should/must/will be improved… 😁
Most of the questions addressed by people could be solved with a better approach.

We started thinking about the whole refactoring last week.
Main idea is to split it into small chunks (more readable and digestible) and to provide a logical "path" with a working result at the end.
We want also put advanced usages in a dedicated section and share some "recipes".

We should have a first preview before the end of march.

**Comment by olets** [comment]
Sounds great! Tangential but if you have favorite examples of usable documentation I'd be interested to hear them

**Comment by xavierfoucrier** [comment]
Thanks for the PR. This will be fixed in the next documentation review #509.
